File: src/entity/Groupe.php
<?php

namespace App\entity;

use App\entity\Visibilite;

use \DateTime;

class Groupe
{
    private string $id;
    private string $nom;
    private string $description;
    private string $code;
    private string $visibilite;
    private int $admin_id;
    private string $created_at;

    private User $admin;
    //constructeur
    public function __construct() {}
    // GETTERS
    public function getId(): string
    {
        return $this->id;
    }
    public function getNom(): string
    {
        return $this->nom;
    }
    public function getDescription(): string
    {
        return $this->description;
    }
    public function getDateCreation(): DateTime
    {
        return Datetime::createFromFormat("jj/mm/aaaa", $this->created_at);
    }
    public function getCode(): string
    {
        return $this->code;
    }
    public function getVisibilite(): string
    {
        // a gerer plus tard
        return $this->visibilite;
    }

    public function getAdmin(): User
    {
        return $this->admin;
    }
    public function getAdminId(): int
    {
        return $this->admin_id;
    }

    // SETTERS
    public function setId(string $id)
    {
        $this->id = $id;
    }
    public function setNom(string $nom)
    {
        $this->nom = $nom;
    }
    public function setDescription(string $description)
    {
        $this->description = $description;
    }
    public function setCode(string $code)
    {
        $this->code = $code;
    }
    public function setVisibilite(string $visibilite)
    {
        $this->visibilite = $visibilite;
    }
    public function setAdmin(User $user)
    {
        $this->admin = $user;
    }
    public function setAdminId(int $id): void
    {
        $this->admin_id = $id;
    }
    public function toDbArray()
    {
        return [
            "nom" => $this->nom,
            "description" => $this->description,
            "code" => $this->code,
            "visibilite" => $this->visibilite,
            "admin_id" => $this->admin_id
        ];
    }
}

File: java/src/entity/Groupe.java
package entity;

import utils.Date;
import java.util.ArrayList;

public class Groupe {
    private String nom;
    private String description;
    private Date dateCreation;
    private User admin;
    private String codeGroupe;
    private ArrayList<Membre> membres;
    private Visibilite visibilite;

    public Groupe(String nom, String description, Date dateCreation) {
        this.nom = nom;
        this.description = description;
        this.dateCreation = dateCreation;
        this.membres = new ArrayList<>();
    }

    public Groupe(String nom, String description, Date dateCreation, User admin, String codeGroupe) {
        this(nom, description, dateCreation);
        this.admin = admin;
        this.codeGroupe = codeGroupe;
    }

    // Getters
    public String getNom() {
        return nom;
    }

    public String getDescription() {
        return description;
    }

    public Date getDateCreation() {
        return dateCreation;
    }

    public User getAdmin() {
        return admin;
    }

    public String getCodeGroupe() {
        return codeGroupe;
    }

    public ArrayList<Membre> getMembres() {
        return membres;
    }

    public int getNbrMembre() {
        return membres.size();
    }

    public Visibilite getVisibilite() {
        return visibilite;
    }

    // Setters
    public void setNom(String nom) {
        this.nom = nom;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setDateCreation(Date dateCreation) {
        this.dateCreation = dateCreation;
    }

    public void setAdmin(User admin) {
        this.admin = admin;
    }

    public void setCodeGroupe(String codeGroupe) {
        this.codeGroupe = codeGroupe;
    }

    public void setVisibilite(Visibilite visibilite) {
        this.visibilite = visibilite;
    }

    public void addMembre(Membre m) {
        m.setGroupe(this);
        membres.add(m);
    }

    public boolean estMembre(User u) {
        for (Membre m : membres) {
            if (m.getUser().getNomUtilisateur().equals(u.getNomUtilisateur())) {
                return true;
            }
        }
        return false;
    }

    @Override
    public String toString() {
        return "Nom=" + nom + ", Description=" + description +
                ", DateCreation=" + dateCreation.toString() +
                ", Admin=" + (admin != null ? admin.getNom() : "N/A") + "}";
    }
}
